src/msg_recorder/recorder.cc

```diff
@@ -23,6 +35,43 @@ MessageRecorder::~MessageRecorder() {
         LOG(INFO) << "Record dir " << GetRecordDir() << " is empty, removing...";
         std::filesystem::remove(GetRecordDir());
     }
+
+    if (std::filesystem::is_empty(GetRecordSnapshotDir())) {
+        LOG(INFO) << "Record Snapshot dir " << GetRecordSnapshotDir() << " is empty, removing...";
+        std::filesystem::remove(GetRecordSnapshotDir());
+    }
+}
+
+void MessageRecorder::GenerateSnapshot(const int& keep_max) {
+    data_copied_ = false;
+    std::unique_lock lock(recorder_mtx_);
+    snapshot_cv_.wait(lock, [this] { return dataflow_paused; });
```

Comment:
Locking is not enough. You pause the dataflow, but there are no guarantee that the written data has flushed to disk or being commited to the DB. You need to properly close the DB before copying, and then reopen it for further recording.

Reply:
I tried with the DB open and close implementation and here are a few issues:
I started the `GenerateSnapshot` function with:
```
    for(const auto& file : files_){
    	// added a file path getter in the RecordFile class
        if (!file->GetFilePath().empty()) {
            std::filesystem::remove_all(file->GetFilePath());
        }
    }
    files_.clear();
```
So from my understanding, all symlinks should be removed and all `RecordFile`s should be destructed through the vector clear method, thus all DB should be closed.

I defined a new struct and a vector `record_init_map` in the header file to save all record data, so I can create all the `RecordFile`s back.

At the end of the function, I did:
```
    for(const auto& data : record_init_map){
        CreateFile(data.message_type_name, data.subid, data.alias);
    }
```
So that all DBs are re-inited and saved in the vector.

Apparently this didn't pass the test, any help would be appreciated, thanks a lot.

Reply:
单就你的error msg而言，我猜是没有create parent dir？

Reply:
看起来因为你的recordfile试图重新生成一个路径，而不是用之前关闭的db的路径。

Reply:
I guess it's because I accidentally remove the path, not the symlink (before I reinitialized the DB back), usually how do we remove a symlink? @xkszltl @chenhao94